content/docs/rfcs/35/README.md

```diff
@@ -0,0 +1,336 @@
+- a Noise handhshake message and/or a Noise transport message;
+- a `ChaChaPoly` ciphertext.
+
+The following fields are concatenated to form the `payload` field:
+
+ - `protocol-id`: identifies the protocol or primitive in use (1 byte). 
+ Supported values are:
+     - `0`:  protocol specification omitted (set for [after-handshake](#After-handshake) messages);
+     - `10`: Noise protocol `Noise_K1K1_25519_ChaChaPoly_SHA256`;
+     - `11`: Noise protocol `Noise_XK1_25519_ChaChaPoly_SHA256`;
+     - `12`: Noise protocol `Noise_XX_25519_ChaChaPoly_SHA256`;
+     - `13`: Noise protocol `Noise_XXpsk0_25519_ChaChaPoly_SHA256`;
+     - `30`: `ChaChaPoly` symmetric encryption.
+ - `handshake-message-len`: the length in bytes of the Noise handshake message  (1 byte). 
+ If `protocol-id` is not equal to `0`, `10`, `11`, `12`, `13`, this field MUST be set to `0`;
+ - `handshake-message`: the Noise handshake message (`handshake-message-len` bytes). 
+If `handshake-message-len` is not `0`, 
+it contains the concatenation of one or more Noise Diffie-Hellman ephemeral or static keys 
+encoded as in [Public Keys Encoding](#Public-Keys-Encoding);
+ - `transport-message-len-len`: the length in bytes of `transport-message-len` (1 byte);
+ - `transport-message-len`: the length in bytes of `transport-message` (`transport-message-len-len` bytes);
+ - `transport-message`: the transport message (`transport-message-len` bytes);
+ - `transport-message-auth`: the symmetric encryption authentication data for `transport-message` (16 bytes).
+
+
+### ABNF
+
+Using [Augmented Backus-Naur form (ABNF)](https://tools.ietf.org/html/rfc5234) we have the following format:
+
+```abnf
+; protocol ID
+protocol-id                 = 1OCTET
+
+; contains the size of handshake-message
+handshake-message-len       = 1OCTET
+
+; contains one or more Diffie-Hellman public keys
+handshake-message           = *OCTET
+
+; contains the size of message-len
+transport-message-len-len   = 1OCTET
+
+; contains the size of transport-message
+transport-message-len       = *OCTET
+
+; contains the transport message, eventually encrypted
+transport-message           = *OCTET
+
+; contains authentication data for transport-message, if encrypted
+transport-message-auth        = 16OCTET
+
+; the Waku WakuMessage payload field
+payload  = protocol-id handshake-message-len handshake-message transport-message-len-len transport-message-len transport-message transport-message-auth 
+```
+
+### Protocol Payload Format
+
+Based on the specified `protocol-id`, 
+the Waku message `payload` field will encode different types of protocol-dependent messages. 
+
+In particular, if `protocol-id` is
+- `0`:  payload encodes an [after-handshake](#After-handshake) message.  
+    - `handshake-message-len` MAY be 0;
+    -  `transport-message` contains the Noise transport message;
+- `10`,`11`,`12`,`13`: payload encodes a supported Noise handshake message.  
+    -  `transport-message` contains the Noise transport message;
```

Comment:
Does the transport message (for the protocol ids 10-13) correspond to the handshake payload explained in the noise framework?
> A handshake message consists of some DH public keys followed by a **payload**. The payload may contain certificates or other data chosen by the application.

Reply:
During an handshake yes, and such payload is protected with a different level of security depending on the handshake round. After an handshake you may simply encrypt payload (which carries your meaningful encrypted messages) with encryption keys, or continue exchanging public keys (ephemeral) along with payloads, or transport messages. Unfortunately payload is a bit confusing with respect to Waku Message payloads. 

Note that a payload/transport-message during/after an handshake is always authenticated, i.e. even if empty, transport-message-auth will contain authentication data for an empty plaintext.

Reply:
You can keep using the transport message term to avoid confusion with the waku payload though, however, I think it is worth mentioning that the transport message for those protocols ids 10-13 corresponds to the handshake payload in the noise framework.

Reply:
Clarified in https://github.com/vacp2p/rfc/commit/505e4055ca5537650853d4fd33e7f8bea33650c0

Reply:
Thanks